refactor: run Miniredis in a non-default port

MikelAlejoBR opened the pull request:
Apparently Miniredis is causing some issues with people who are running a full Redis instance locally. It seems that turning off the Redis instance makes the tests pass again, so changing the port on which Miniredis runs should do the trick.

Instead of using the "RunT" function of Miniredis and requesting a random port, a hardcoded one is used instead. This is to avoid having the one in a million chance of getting the default Redis port as the random port. Hopefully this will avoid head scratching to the developer who has to experience tests failing once, and then passing without any issue in the next run.

MikelAlejoBR committed:
refactor: run Miniredis in a non-default port

Apparently Miniredis is causing some issues with people who are running
a full Redis instance locally. It seems that turning off the Redis
instance makes the tests pass again, so changing the port on which
Miniredis runs should do the trick.

Instead of using the "RunT" function of Miniredis and requesting a
random port, a hardcoded one is used instead. This is to avoid having
the one in a million chance of getting the default Redis port as the
random port. Hopefully this will avoid head scratching to the developer
who has to experience tests failing once, and then passing without any
issue in the next run.

lindgrenj6 commented on redis/marketplace_token_test.go:
```diff
@@ -28,7 +29,8 @@ func setUpFakeToken() *marketplace.BearerToken {
 // TestGetTokenBadTenant tests that when given a bad or nonexistent tenant, an expected error is returned.
 func TestGetTokenBadTenant(t *testing.T) {
-	mr, err := miniredis.Run()
+	mr := miniredis.NewMiniRedis()
```
So this is probably a weird request - but could we spin this up in the `TestMain` package that way it's there for all the tests? Or is there a reason we're spinning up a new one every time?

MikelAlejoBR replied:
Yes, that is a good idea. I initially tried to do that, but I faced some problems I thought they were caused by Miniredis being run at the package level. But it seems that it was something else. Implemented the start/stop at the package level. Thanks!

MikelAlejoBR committed:
refactor: start and stop the Miniredis instance at the package level

Instead of having to start and stop Miniredis every time, which consumes
resources and bloats the code, the instance can be run at the package
level and leave to the tester the responsability of creating a new
instance if they so desire.

It is true that the state is shared among the tests, but for now this is
not a problem.
